File: app/app/src/main/java/io/github/homework/MainActivity.java
package io.github.homework;

import androidx.appcompat.app.AppCompatActivity;
import androidx.viewpager.widget.ViewPager;

import android.content.ContentValues;
import android.content.Intent;
import android.database.sqlite.SQLiteDatabase;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.google.android.material.tabs.TabLayout;

import java.util.ArrayList;
import java.util.List;

import io.github.homework.adapter.PagerAdapter;
import io.github.homework.constant.Constant;
import io.github.homework.database.MyDatabaseHelper;
import io.github.homework.entity.Music;

public class MainActivity extends AppCompatActivity {

    private TabLayout mTabLayout;
    private ViewPager mViewPager;

    private PagerAdapter pagerAdapter;
    private List<Music> musicList;
    private List<String> titleList;
    private List<Integer> typeList;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initDatabase();

        mTabLayout = findViewById(R.id.tab_layout);
        mViewPager = findViewById(R.id.view_pager);


        titleList = new ArrayList<>();
        titleList.add(Constant.FIRST_NAME);
        titleList.add(Constant.SECOND_NAME);

        typeList = new ArrayList<>();
        typeList.add(Constant.TYPE_FIRST);
        typeList.add(Constant.TYPE_SECOND);


        for (String name : titleList) {
            mTabLayout.addTab(mTabLayout.newTab().setText(name));
        }
        pagerAdapter = new PagerAdapter(getSupportFragmentManager(), titleList, typeList);

        mViewPager.setAdapter(pagerAdapter);
        mTabLayout.setupWithViewPager(mViewPager);
        mTabLayout.setTabIndicatorFullWidth(false);
    }

    /**
     * 初始化数据库
     */
    private void initDatabase() {
        musicList = new ArrayList<>();
        musicList.clear();
        musicList.add(new Music("Stairway", "Patrick Patrikios", R.drawable.cover01, R.raw.stairway,1));
        musicList.add(new Music("Awful", "josh pan", R.drawable.cover02, R.raw.awful,1));
        musicList.add(new Music("Voices", "Patrick Patrikios", R.drawable.cover03, R.raw.voices,1));
        musicList.add(new Music("Breatha", "josh pan", R.drawable.cover04, R.raw.breatha,1));
        musicList.add(new Music("19th Floor", "Bobby Richards", R.drawable.cover05, R.raw.floor,1));

        musicList.add(new Music("袖手旁观", "张宇", R.drawable.cover05, R.raw.chinesesong01,2));
        musicList.add(new Music("暗香", "沙宝亮", R.drawable.cover01, R.raw.chinesesong02,2));
        musicList.add(new Music("无地自容", "黑豹乐队", R.drawable.cover02, R.raw.chinesesong03,2));
        musicList.add(new Music("吻的太逼真", "张敬轩", R.drawable.cover05, R.raw.chinesesong04,2));
        musicList.add(new Music("春夏秋冬", "张国荣", R.drawable.cover05, R.raw.chinesesong05,2));


        MyDatabaseHelper dbHelper = new MyDatabaseHelper(this, "MusicData.db", null, 1);
        // 创建数据库
        dbHelper.getWritableDatabase();
        SQLiteDatabase mSqlitedb = dbHelper.getWritableDatabase();
        // 清空数据库
        mSqlitedb.execSQL("DELETE FROM Music");
        // 插入预设数据
        for (int i = 0; i < musicList.size(); i++) {
            Music music = musicList.get(i);
            ContentValues values = new ContentValues();
            values.put("name", music.getName());
            values.put("author", music.getAuthor());
            values.put("cover", music.getCover());
            values.put("resrouce", music.getResource());
            values.put("type",music.getType());
            mSqlitedb.insert("Music", null, values);
        }
    }
}

File: storagetest/app/src/main/java/io/github/storagetest/MainActivity.kt
package io.github.storagetest

import android.content.ContentValues
import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*
import java.io.*
import java.lang.StringBuilder

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val dbHelper = MyDatabaseHelper(this, "BookStore.db", 1)
        // 创建数据库
        create_table.setOnClickListener {
            dbHelper.writableDatabase
        }
        // 增加数据
        add_data.setOnClickListener {
            val db = dbHelper.writableDatabase
            val values1 = ContentValues().apply {
                put("name", "admin")
                put("author", "me")
                put("pages", 252)
                put("price", 16.96)
            }
            db.insert("Book", null, values1)
            val values2 = ContentValues().apply {
                put("name", "root")
                put("author", "you")
                put("pages", 454)
                put("price", 96.16)
            }
            db.insert("Book", null, values2)
        }
        // 更新数据
        update_data.setOnClickListener {
            Toast.makeText(this, "更新数据", Toast.LENGTH_SHORT).show()
            val db = dbHelper.writableDatabase
            val values = ContentValues()
            values.put("price", 19.99)
            db.update("Book", values, "name = ?", arrayOf("root"))
        }
        // 删除数据
        delete_data.setOnClickListener {
            Toast.makeText(this, "删除数据", Toast.LENGTH_SHORT).show()
            val db = dbHelper.writableDatabase
            db.delete("Book", "pages > ?", arrayOf("400"))
        }
        // 查询数据
        query_data.setOnClickListener {
            Toast.makeText(this, "查询数据", Toast.LENGTH_SHORT).show()
            val db = dbHelper.writableDatabase

            val cursor = db.rawQuery("select * from Book", null)
            if (cursor.moveToNext()) {
                do {
                    val name = cursor.getString(cursor.getColumnIndex("name"))
                    val author = cursor.getString(cursor.getColumnIndex("author"))
                    val pages = cursor.getString(cursor.getColumnIndex("pages"))
                    val price = cursor.getString(cursor.getColumnIndex("price"))
                    Log.d("==>", "查询到的数据是：$name,$author,$pages,$price")
                } while (cursor.moveToNext())
            }
            cursor.close()
        }
    }
}